### PR TITLE
[WIP] Rails 5 release notes

### PR DESCRIPTION
**Pull request by maclover7**
[ci skip]

Not ready for merge yet. Drafting up release notes using 4.0 as a template. Ref #20612.

### CODE REVIEW

**Comment by connorshea** [comment]
Is "exclusive use of rails CLI over rake" an accurate statement? I was under the impression that not everything had been translated over to equivalent rails commands yet.

**Comment by connorshea** [comment]
Typo on line 122, "defintion"

File: guides/source/5_0_relase_notes.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+### Upgrade
+
+* **Ruby 2.2.2** ([commit](https://github.com/rails/rails/commit/d3b098b8289ffaa8486f526dc53204123ed581f3)) - Ruby 2.2.2+ required
+* **Rails API** ([commit](https://github.com/rails/rails/pull/19832)) - Rails API is merged directly into Rails
+* **Make ActionController::Parameters not inherited from Hash** ([Pull Request](https://github.com/rails/rails/pull/20868))
+* **Sprockets 3 is out** ([Upgrading Guide](https://github.com/rails/sprockets/blob/master/UPGRADING.md))
+
+### General
+
+* **Remove debugger supprt** ([commit](https://github.com/rails/rails/commit/93559da4826546d07014f8cfa399b64b4a143127)) - Byebugger doesn't work with Ruby 2.2, so it is incompatible with Rails 5.0.
```
Comment (egilburg) [inline guides/source/5_0_relase_notes.md]: I think it's `debugger` which doesn't work with ruby 2.2, while `byebug` works fine.

File: guides/source/5_0_relase_notes.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+--------------------------------------------------------------------------------
+
+Upgrading to Rails 5.0
+----------------------
+
+If you're upgrading an existing application, it's a great idea to have good test coverage before going in. You should also first upgrade to Rails 4.2 in case you haven't and make sure your application still runs as expected before attempting an update to Rails 5.0. A list of things to watch out for when upgrading is available in the [Upgrading Ruby on Rails](upgrading_ruby_on_rails.html#upgrading-from-rails-4-2-to-rails-5-0) guide.
+
+
+Creating a Rails 5.0 application
```
Comment (senny) [inline guides/source/5_0_relase_notes.md]: Can we remove this section? I think this was added to the release notes when Rails fully adopted bundler (and maybe in a few releases after that). Bundler has been around for a long enough time.

File: guides/source/5_0_relase_notes.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+If you have a local checkout of the Rails repository and want to generate an application using that, you can pass the `--dev` flag:
+
+```
+$ ruby /path/to/rails/railties/bin/rails new myapp --dev
+```
+
+Major Features
+--------------
+
+TODO: Add brain map? -- 4.0 release had one (see below)
```
Comment (senny) [inline guides/source/5_0_relase_notes.md]: It's not necessary for us to include such an image. If I remember correctly that was provided by someone writing a blog post about Rails 4. Let's remove the todo and the image for now.

**Comment by maclover7** [comment]
Updated. Will squash down later on.

**Comment by senny** [comment]
@maclover7 would you mind squashing right away? I'd like to get the foundation merged soon so that we can iterate on it without having everything in PR comments. Does that work for you?

**Comment by maclover7** [comment]
Sure thing -- I'll squash down as soon as I get home from school later today. :)

**Comment by maclover7** [comment]
squashed and ready to be merged! @senny

**Comment by senny** [comment]
@maclover7 thank you :yellow_heart: I'll go through the changelogs today and pick append relevant changes.

**Comment by maclover7** [comment]
:+1:

**Pull request merged**